### ISSUE TITLE
Don't use shell=True in subprocess

### ISSUE THREAD

**Issue by MasterOdin** [asciidoc-py/asciidoc-py#24]
This is to separate the discussion of this from #20, as well as track possible work on this (given that #20 is probably going to be merged before this is done).

asciidoc implements being able to use external filters/code to augment the central asciidoc process. However, it does this by using `subprocess.Popen` and by setting `shell=True`. Asciidoc handles putting together the arguments to the program as well as trying to handle properly quoting things. The benefit here is that theoretically, it allows asciidoc to utilize shell builtins (like globing and environment variables).

However, the downside is that it opens itself up to potential attack vectors. An example is this filter tag: `[music,slidy-example__1.png" -; rm -rf /tmp/test #]` which would ends up getting executed as: `'"/usr/local/opt/python/bin/python3.6" "/Users/mpeveler/Github/asciidoc3/filters/music/music2png.py" -m -o "/Users/mpeveler/Github/asciidoc3/doc/slidy-example__1.png" -; rm -rf /tmp/test #" -'`

I'm able to get the music filter to pass as one would expect, but attach an attack (`rm -rf /tmp/test`) to be secondarily executed off the Popen. I'm not sure the ability to use glob or environment variables in these execution statements is worth this security hole, or even trying to do "proper" quoting. In short, just use `shell=False` and let python handle the quoting of arguments for us.

**Comment by elextr** [comment]
> In short, just use shell=False and let python handle the quoting of arguments for us.

Yes, Python knows the correct quoting rules for the platform its running on.
